### ISSUE TITLE
Unable to create message resource with MediaUrl(s)

### ISSUE THREAD

**Issue by dprothero** [twilio/twilio-csharp#294]
**Version:**
From next-gen branch: https://github.com/twilio/twilio-csharp/commit/fdb76b2bd71afbd10847c29d47855faa82b7af83
**API Subdomain (api/taskrouter/ip_messaging):**
Twilio.Rest.Api.V2010.Account

### Code Snippet
```c#
Twilio.TwilioClient.Init(
    Environment.GetEnvironmentVariable("TWILIO_ACCOUNT_SID"),
    Environment.GetEnvironmentVariable("TWILIO_AUTH_TOKEN"));

var creator = MessageResource.Creator(
    new PhoneNumber("+15005551111"),
    new PhoneNumber("+15005552222"),
    "What's up?");
creator.MediaUrl = new List<Uri>()
{
    new Uri("https://c1.staticflickr.com/3/2899/14341091933_1e92e62d12_b.jpg")
};
var message = creator.Create();
Debug.WriteLine(message.Sid);
```

### Exception / Log
```
Twilio.Exceptions.RestException was unhandled
  Body=Media urls: System.Collections.Generic.List`1[System.Uri] are invalid. Please use only valid http and https urls
  Code=21620
  HResult=-2146233088
  Message=Exception of type 'Twilio.Exceptions.RestException' was thrown.
  MoreInfo=https://www.twilio.com/docs/errors/21620
  Source=Twilio
  Status=400
  StackTrace:
       at Twilio.Http.WebRequestClient.<>c.<MakeRequest>b__1_0(Exception x) in C:\Users\david\Projects\twilio-csharp\Twilio\Http\WebRequestClient.cs:line 74
       at System.AggregateException.Handle(Func`2 predicate)
       at Twilio.Http.WebRequestClient.MakeRequest(Request request) in C:\Users\david\Projects\twilio-csharp\Twilio\Http\WebRequestClient.cs:line 67
       at Twilio.Clients.TwilioRestClient.Request(Request request) in C:\Users\david\Projects\twilio-csharp\Twilio\Clients\TwilioRestClient.cs:line 67
       at Twilio.Rest.Api.V2010.Account.MessageCreator.Create(ITwilioRestClient client) in C:\Users\david\Projects\twilio-csharp\Twilio\Rest\Api\V2010\Account\MessageCreator.cs:line 139
       at Twilio.Base.Creator`1.Create() in C:\Users\david\Projects\twilio-csharp\Twilio\Base\Creator.cs:line 33
       at ConsoleApplication1.Program.Main(String[] args) in C:\Users\david\Projects\twilio-csharp\ConsoleApplication1\Program.cs:line 28
       at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ThreadHelper.ThreadStart()
  InnerException: 
```

### Steps to Reproduce
Try to send an MMS message by setting the MediaUrl property of the MessageCreator to any valid URL of an image. I've tried multiple good URL's.

**Comment by jmonasterio** [comment]
Same problem here. I called a little differently:

```
TwilioClient.Init(accountSid, authToken);
            
            var uriList = new List<Uri>() { new Uri("https://www.google.com/images/nav_logo242.png") };

            var task = 
            new MessageCreator(
              toPhone,
              sendingPhone,
              uriList
            ).CreateAsync(restClient);

            var message = await task;

            Console.WriteLine(message.sid);
```

**Comment by dprothero** [comment]
The issue has been fixed in v5.0.0-rca2. The library has a new interface. Here's the working code:

```c#
using System;
using System.Collections.Generic;
using System.Diagnostics;
using Twilio;
using Twilio.Rest.Api.V2010.Account;
using Twilio.Types;

namespace ConsoleApplication14
{
    class Program
    {
        static void Main(string[] args)
        {
            TwilioClient.Init(
                Environment.GetEnvironmentVariable("TWILIO_ACCOUNT_SID"),
                Environment.GetEnvironmentVariable("TWILIO_AUTH_TOKEN")
            );

            var message = MessageResource.Create(
                from: new PhoneNumber("+12093373517"), 
                to: new PhoneNumber("+12099864420"),
                body: "What's up?",
                mediaUrl: new List<Uri>()
                {
                    new Uri("https://c1.staticflickr.com/3/2899/14341091933_1e92e62d12_b.jpg")
                }
            );

            Debug.WriteLine(message.Sid);
        }
    }
}
```
